packages/validator/src/processor.ts

```diff
@@ -0,0 +1,204 @@
+import * as utils from './utils/utils';
+import * as logger from './utils/logger';
+import { validateCondition } from './validators/conditionValidator';
+import { validateRegexp } from './validators/regexpValidator';
+import { validateOptional } from './validators/optionalValidator';
+import { isAnyParamValid } from './validators/anyValidator';
+import { Log, Result, Roots } from './types';
+import { execute } from './utils/action';
+import fs from 'fs';
+
+const apiTemplate = JSON.parse(fs.readFileSync('templates/apiSpecifications.json', 'utf8'));
+const oisTemplate = JSON.parse(fs.readFileSync('templates/ois.json', 'utf8'));
+const endpointsTemplate = JSON.parse(fs.readFileSync('templates/endpoints.json', 'utf8'));
+
+/**
+ * Recursion validating provided specification against template
+ * @param specs - specification that is being validated
+ * @param template - template the specification is validated against
+ * @param paramPath - string of parameters separated by ".", representing path to current specs location (empty string in root)
```

Comment:
nit: spelling error: "empty string _is_ root"

Reply:
Fixed in 2b2777b